core/src/main/java/com/linecorp/armeria/server/RoutingPredicate.java

```diff
@@ -83,7 +83,8 @@
         requireNonNull(headersPredicate, "headersPredicate");
         final Matcher m = CONTAIN_PATTERN.matcher(headersPredicate);
         if (m.matches()) {
-            final Predicate<HttpHeaders> predicate = headers -> headers.contains(m.group(2));
+            final CharSequence headerName = m.group(2);
```

Comment:
How about converting to `AsciiString` using `HttpHeaderNames.of()`, so that `HttpHeaders.contains()` doesn't have to do the conversion internally every time?